### ISSUE TITLE
[🐞] Cannot run docs package locally - dependencies not built

### ISSUE THREAD

**Issue by Balastrong** [BuilderIO/qwik#5743]
### Which component is affected?

Starters / CLI

## Overview
The goal was to run locally the docs package, so as per CONTRIBUTING.md I run `pnpm install` followed by `pnpm docs.dev`.

The second command failed because the package `qwik-city` is not built.

I managed to run the docs with some extra steps (and the help of the amazing @gioboa) and I'm reporting here what I did.

I think the outcome could be either updating the docs with the extra steps or updating the scripts to no longer require those steps.

(see the "Additional Information" section here below for all the process)

## Summary:
Some packages are not built by the current scripts, so if you try to run `pnpm docs.dev` they are still missing and have to be built manually.

To make it work, I did in order:
1. `pnpm install`
2. `pnpm build.eslint`
3. `pnpm build`
4. `cd packages/qwik-react/; pnpm build`
5. `pnpm docs.dev` 

### Reproduction

-

### Steps to reproduce

1. Clone the repo
2. `pnpm install`
3. `pnpm docs.dev`

### System Info

```shell
System:
    OS: macOS 14.2.1
    CPU: (8) arm64 Apple M1
    Memory: 48.84 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.11.0 - ~/.nvm/versions/node/v20.11.0/bin/node
    Yarn: 1.22.21 - ~/.nvm/versions/node/v18.18.2/bin/yarn
    npm: 10.2.4 - ~/.nvm/versions/node/v20.11.0/bin/npm
    pnpm: 8.14.1 - ~/.nvm/versions/node/v20.11.0/bin/pnpm
  Browsers:
    Chrome: 120.0.6099.234
    Safari: 17.2.1
  npmPackages:
    @builder.io/partytown: ^0.8.2 => 0.8.2 
    @builder.io/qwik: workspace:* => 1.4.0 
    undici: * => 6.3.0 
    vite: ^5.0.11 => 5.0.11
```

#### Step 1
After the first install + docs.dev run, the error is:
```
failed to load config from /Users/leonardo/PersProgetti/qwik/packages/docs/vite.config.ts
error when starting dev server:
Error: Cannot find module '/Users/leonardo/PersProgetti/qwik/packages/docs/node_modules/@builder.io/qwik-city/lib/vite/index.cjs'
    at createEsmNotFoundErr (node:internal/modules/cjs/loader:1181:15)
```

#### Step 2
However, if I try to run `pnpm build` I get this error
```
 ELIFECYCLE  Command failed with exit code 2.

undefined

 ELIFECYCLE  Command failed with exit code 1.

❌ Error: Command failed with exit code 1: pnpm build

Oops! Something went wrong! :(

ESLint: 8.56.0

ESLint couldn't find the plugin "eslint-plugin-qwik".

(The package "eslint-plugin-qwik" was not found when loaded as a Node module from the directory "/Users/leonardo/PersProgetti/qwik/packages/qwik-labs".)

It's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm install eslint-plugin-qwik@latest --save-dev

The plugin "eslint-plugin-qwik" was referenced from the config file in ".eslintrc.cjs".
```

#### Step 3
If I run `pnpm build.eslint` and then `pnpm build` it works.

This probably built the missing `eslint-plugin-qwik` required by the build.

#### Step 4
If I try again to run `pnpm docs.dev` I get a new error, this time the culprit is `qwik-react` which seemingly hasn't been built by the previous commands.

```
failed to load config from /Users/leonardo/PersProgetti/qwik/packages/docs/vite.config.ts
error when starting dev server:
Error: Cannot find module '/Users/leonardo/PersProgetti/qwik/packages/docs/node_modules/@builder.io/qwik-react/lib/vite.cjs'
    at createEsmNotFoundErr (node:internal/modules/cjs/loader:1181:15)
```

#### Step 5
I can manually build `qwik-react` with
```
cd packages/qwik-react/
pnpm build
```

#### Step 6
Going back to the root, pnpm `doc.dev` now works!

**Comment by wmertens** [comment]
What was missing was build.core, the docs scripts should do a check if the core build exists and if not run it

**Comment by balaji-sivasakthi** [comment]
@wmertens, I'm really interested in starting to contribute by fixing this issue.

**Comment by wmertens** [comment]
Fixed via #5755
